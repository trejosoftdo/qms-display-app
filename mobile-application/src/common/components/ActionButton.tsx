import { FC } from 'react';
import { StyleSheet } from 'react-native';
import { Button } from 'react-native-paper';

/**
 * ActionButtonProps defines the props for the Action Button Component.
 *
 * @property onPress - a handler for the on press event
 * @property message - the button display message
 * @property icon - the button icon
 */
interface ActionButtonProps {
  onPress: () => void;
  message: string;
  icon: string;
}

/**
 * A component that displays a button with icon
 *
 * @param {ActionButtonProps} props - The props for the Action Button component.
 */
const ActionButton: FC<ActionButtonProps> = (props: ActionButtonProps) => (
  <Button
    style={styles.container}
    labelStyle={styles.label}
    mode="contained"
    onPress={props.onPress}
    icon={props.icon}
  >
    {props.message}
  </Button>
);

const styles = StyleSheet.create({
  container: {
    borderRadius: 8,
  },
  label: {
    fontSize: 16,
    lineHeight: 24
  },
});

export default ActionButton;
