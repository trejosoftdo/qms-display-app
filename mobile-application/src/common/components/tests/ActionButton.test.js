import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import ActionButton from '../ActionButton';


describe('ActionButton component', () => {
  const props = {
    icon: 'lock',
    onPress: jest.fn(),
    message: 'Test message',
  };

  it('renders the component with the provided props as expected', () => {
    expect(renderShallow(<ActionButton {...props} />)).toMatchSnapshot();
  });

  it('calls the onPress handler when being clicked', () => {
    const { getByText } = render(<ActionButton {...props} />);
    const element = getByText(props.message);
    fireEvent.press(element);
    expect(props.onPress).toHaveBeenCalled();
  });
});
