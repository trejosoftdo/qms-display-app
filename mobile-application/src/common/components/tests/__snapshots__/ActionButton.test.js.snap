// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionButton component renders the component with the provided props as expected 1`] = `
<Button
  icon="lock"
  labelStyle={
    {
      "fontSize": 16,
      "lineHeight": 24,
    }
  }
  mode="contained"
  onPress={[MockFunction]}
  style={
    {
      "borderRadius": 8,
    }
  }
>
  Test message
</Button>
`;
